docs/get_started/tour.mdx

```diff
@@ -1473,6 +1473,35 @@ npm run part4
 </Tabs>
 </Admonition>
 
+## Idempotency for any request
+As you saw, generating idempotency keys inside your handlers and storing them in Restate is easy.
+
+But this doesn't guard us yet against retries of the HTTP request to Restate.
+If the system invoking the `checkout` handler retries the request, the handler gets executed twice.
+
+To cover this, you can add an `idempotency-key` header to the incoming request to let Restate deduplicate them.
+
+Our `CartObject/checkout` handler is robust against retries since it checks the state content.
+So instead, check out how this works by calling the `CheckoutService/handle` handler with an idempotency key:
```

Comment:
I like the introduction of idempotency. So far the user has always interacted with the app through the `CartObject`, right? I am wondering whether it would feel more natural to stick with this service because also for `CartObject/checkout` and idempotent invocation will have benefits. I am thinking about the scenario where the caller will do something based on the return value of `CartObject/checkout`. If one calls this method multiple times, it will only return `true` on the first successful run, I believe.

The reason what made me think about this is that `CheckoutService` feels like an internal service that is not invoked from externally.

Reply:
Okay, I'll do it like this